vercel/resource_shared_environment_variable_project_link.go

```diff
@@ -0,0 +1,224 @@
+	TeamID                      types.String `tfsdk:"team_id"`
+}
+
+func (e *SharedEnvironmentVariableProjectLink) toUpdateSharedEnvironmentVariableRequest(ctx context.Context, link bool) (req client.UpdateSharedEnvironmentVariableRequest, ok bool) {
+	upd := client.UpdateSharedEnvironmentVariableRequestProjectIDUpdates{}
+
+	if link {
+		upd.Link = []string{e.ProjectID.ValueString()}
+	} else {
+		upd.Unlink = []string{e.ProjectID.ValueString()}
+	}
+
+	return client.UpdateSharedEnvironmentVariableRequest{
+		TeamID:           e.TeamID.ValueString(),
+		EnvID:            e.SharedEnvironmentVariableID.ValueString(),
+		ProjectIDUpdates: upd,
+	}, true
+}
+
+func (r *sharedEnvironmentVariableProjectLinkResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan SharedEnvironmentVariableProjectLink
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	request, ok := plan.toUpdateSharedEnvironmentVariableRequest(ctx, true)
+	if !ok {
+		return
+	}
+
+	response, err := r.client.UpdateSharedEnvironmentVariable(ctx, request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error linking project to shared environment variable",
+			"Could not link project to shared environment variable, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := SharedEnvironmentVariableProjectLink{
+		TeamID:                      types.StringValue(response.TeamID),
+		SharedEnvironmentVariableID: types.StringValue(response.ID),
+		ProjectID:                   plan.ProjectID,
+	}
+
+	tflog.Info(ctx, "linked shared environment variable to project", map[string]interface{}{
+		"team_id":                        result.TeamID.ValueString(),
+		"shared_environment_variable_id": result.SharedEnvironmentVariableID.ValueString(),
+		"project_id":                     result.ProjectID.ValueString(),
+	})
+
+	diags = resp.State.Set(ctx, result)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *sharedEnvironmentVariableProjectLinkResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state SharedEnvironmentVariableProjectLink
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	response, err := r.client.GetSharedEnvironmentVariable(ctx, state.TeamID.ValueString(), state.SharedEnvironmentVariableID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading shared environment variable",
+			"Could not read shared environment variable, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := SharedEnvironmentVariableProjectLink{
+		TeamID:                      types.StringValue(response.TeamID),
+		SharedEnvironmentVariableID: types.StringValue(response.ID),
+		ProjectID:                   state.ProjectID,
+	}
+
+	for _, pid := range response.ProjectIDs {
```

Comment:
Any reason this can't be a sliced.Contains(response.ProjectIDs, result.ProjectID.ValueString())?

Reply:
In fact in fact, should probably use state.ProjectID.ValueString and build result after the check.